index.js

```diff
@@ -155,23 +147,35 @@ function createQuestions(config) {
  * Format the git commit message from given answers.
  *
  * @param {Object} answers Answers provide by `inquier.js`
+ * @param {Object} config Result of the `getConfig` returned promise
  * @return {String} Formated git commit message
  */
-function format(answers) {
+
+function format(answers, config) {
+  const format = config.format
+
   const { columns } = process.stdout
 
-  const head = truncate(answers.subject, columns)
+  const emoji = answers.type
+  const type = types.find((type) => type.code === emoji.emoji).name
+  const scope = answers.scope ? '(' + answers.scope.trim() + ')' : ''
+  const subject = answers.subject.trim()
+
+  const commitMessage = format
+    .replace(/\{emoji\}/g, emoji.emoji)
+    .replace(/\{type\}/g, type)
+    .replace(/\{scope\}/g, scope)
+    .replace(/\{subject\}/g, subject)
```

Comment:
Minor, for simplicity I would use `replaceAll` with strings instead:

```suggestion
  const commitMessage = format
    .replaceAll('{emoji}', emoji.emoji)
    .replaceAll('{type}', type)
    .replaceAll('{scope}', scope)
    .replaceAll('{subject}', subject)
```